src/Raven.Server/Documents/ShardedHandlers/ShardedCommands/ShardedBaseCommand.cs

```diff
@@ -17,12 +19,15 @@ public abstract class ShardedBaseCommand<T> : RavenCommand<T>
         public readonly HttpMethod Method;
 
         public HttpResponseMessage Response;
-        
+        private readonly IDisposable _disposable;
+        public TransactionOperationContext Context;
 
         public override bool IsReadRequest => false;
 
         protected ShardedBaseCommand(ShardedRequestHandler handler, Headers headers, BlittableJsonReaderObject content = null)
         {
+            _disposable = handler.ContextPool.AllocateOperationContext(out Context);
```

Comment:
do we need that each time? for all the commands?

Reply:
not needed anymore. done.